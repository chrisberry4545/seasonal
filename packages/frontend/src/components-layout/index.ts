export * from './ImageGridItem/ImageGridItem';

export * from './ImageGrid/ImageGrid';

export * from './SummaryColumn';

export * from './SummaryColumnList';

export * from './SeasonNameView/SeasonNameView';

export * from './Toast/Toast';
