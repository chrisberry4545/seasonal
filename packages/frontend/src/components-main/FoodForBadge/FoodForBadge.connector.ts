import { connect } from 'react-redux';
import {
  FoodForBadge
} from './FoodForBadge';
import {
  selectIsFoodDataOrBasicSeasonsLoading,
  selectVisibleFoodForBadgeDetailsData,
  badgeDetailsSelectFood
} from '@chrisb-dev/seasonal-shared-frontend-redux';

import { IState } from '../../store';
import {
  IFoodForBadgeInputProps,
  IFoodForBadgeDispatchProps
} from './FoodForBadge.interface';
import { Dispatch } from 'redux';

const mapStateToProps = (
  state: IState
): IFoodForBadgeInputProps => ({
  currentBadgeFood: selectVisibleFoodForBadgeDetailsData(state),
  isLoading: selectIsFoodDataOrBasicSeasonsLoading(state)
});

const mapDispatchToProps = (
  dispatch: Dispatch
): IFoodForBadgeDispatchProps => ({
  onFoodSelected: (foodId: string) => (
    dispatch(badgeDetailsSelectFood(foodId))
  )
});

export const FoodForBadgeConnecter = connect(
  mapStateToProps,
  mapDispatchToProps
)(FoodForBadge);
