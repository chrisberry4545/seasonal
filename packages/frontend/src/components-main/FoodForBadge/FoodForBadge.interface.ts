import { IFood } from '@chrisb-dev/seasonal-shared-models';

export interface IFoodForBadgeInputProps {
  isLoading: boolean;
  currentBadgeFood: IFood[] | undefined;
}

export interface IFoodForBadgeDispatchProps {
  onFoodSelected: (foodId: string) => void;
}

export interface IFoodForBadgeProps
  extends IFoodForBadgeInputProps, IFoodForBadgeDispatchProps {}
