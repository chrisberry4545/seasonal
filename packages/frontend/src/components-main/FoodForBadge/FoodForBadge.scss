@import '../../styles/settings';
@import '../../styles/tools';

.c-food-for-badge {
  margin-top: 1.5rem;
  border-bottom: 1px solid $grey-med;
  padding-bottom: 1rem;
  @include mq($from: tablet) {
    padding-bottom: 2rem;
  }
}

  .c-food-for-badge__heading {
    width: 100%;
    text-align: center;
    display: block;
  }
