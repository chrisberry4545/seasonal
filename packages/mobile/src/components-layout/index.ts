export * from './ConfirmButtonLayout/ConfirmButtonLayout';

export * from './DefaultPaddingContainer/DefaultPaddingContainer';

export * from './DetailsPageLayout/DetailsPageLayout';

export * from './FeedbackQuestionsLayout/FeedbackQuestionsLayout';

export * from './HeaderAndBackButton/HeaderAndBackButton';

export * from './ImageGrid/ImageGrid';

export * from './ImageGridItem/ImageGridItem';

export * from './MainContainer/MainContainer';

export * from './ModalLayout/ModalLayout';

export * from './NoResultsText/NoResultsText';

export * from './SeasonNameView/SeasonNameView';

export * from './SwitchableGridOrList/SwitchableGridOrList';

export * from './Toast/Toast';

export * from './TopLoadingSpinner/TopLoadingSpinner';
