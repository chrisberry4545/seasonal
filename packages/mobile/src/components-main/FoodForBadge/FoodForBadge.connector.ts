import { connect } from 'react-redux';
import {
  FoodForBadge
} from './FoodForBadge';
import {
  selectIsCurrentBadgeDetailsLoading,
  selectVisibleFoodForBadgeDetailsData,
  badgeDetailsSelectFood
} from '@chrisb-dev/seasonal-shared-frontend-redux';

import { IState } from '../../store';
import {
  IFoodForBadgeInputProps,
  IFoodForBadgeDispatchProps
} from './FoodForBadge.interface';
import { Dispatch } from 'redux';

const mapStateToProps = (
  state: IState
): IFoodForBadgeInputProps => ({
  currentBadgeFood: selectVisibleFoodForBadgeDetailsData(state),
  isLoading: selectIsCurrentBadgeDetailsLoading(state)
});

const mapDispatchToProps = (
  dispatch: Dispatch
): IFoodForBadgeDispatchProps => ({
  onFoodSelected: (foodId: string) => (
    dispatch(badgeDetailsSelectFood(foodId))
  )
});

export const FoodForBadgeConnecter = connect(
  mapStateToProps,
  mapDispatchToProps
)(FoodForBadge);
