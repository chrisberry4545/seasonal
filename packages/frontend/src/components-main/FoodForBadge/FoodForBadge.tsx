import React, { FC } from 'react';
import './FoodForBadge.scss';
import { IFoodForBadgeProps } from './FoodForBadge.interface';
import { ImageGrid } from '../../components-layout';
import {
  TextMedium
} from '@chrisb-dev/seasonal-shared-frontend-components';

export const FoodForBadge: FC<IFoodForBadgeProps> = ({
  isLoading,
  currentBadgeFood,
  onFoodSelected
}) => (
  !isLoading
    ? <div className='c-food-for-badge'>
      <TextMedium className='c-food-for-badge__heading'>
        Found in
      </TextMedium>
      <ImageGrid data-e2e='food-for-badge-grid'
        data={currentBadgeFood} onClick={onFoodSelected} />
    </div>
    : null
);
