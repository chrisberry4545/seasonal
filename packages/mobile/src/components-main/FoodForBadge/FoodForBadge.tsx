import React, { FC } from 'react';
import { IFoodForBadgeProps } from './FoodForBadge.interface';
import { ImageGrid } from '../../components-layout';
import {
  TextMedium
} from '../../components-elements';
import { View, TextStyle } from 'react-native';

const styleFoodForBadgeFoundIn: TextStyle = {
  marginBottom: 12,
  textAlign: 'center',
  width: '100%'
};

export const FoodForBadge: FC<IFoodForBadgeProps> = ({
  isLoading,
  currentBadgeFood,
  onFoodSelected
}) => (
  !isLoading
    ? <View>
      <TextMedium style={styleFoodForBadgeFoundIn}>
        Found in
      </TextMedium>
      <ImageGrid data-e2e='food-for-badge-grid'
        data={currentBadgeFood} onClick={onFoodSelected} />
    </View>
    : null
);
