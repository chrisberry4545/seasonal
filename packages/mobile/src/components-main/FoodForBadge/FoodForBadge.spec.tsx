import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { FoodForBadge } from './FoodForBadge';
import { IFood } from '@chrisb-dev/seasonal-shared-models';
import { ImageGrid } from '../../components-layout';

jest.mock('../../components-elements', () => ({
  TextMedium: () => 'TextMedium'
}));
jest.mock('../../components-layout', () => ({
  ImageGrid: () => 'ImageGrid'
}));

describe('<FoodForBadge />', () => {
  let wrapper: ShallowWrapper;
  const food = [{}, {}] as IFood[];
  let mockOnFoodSelected: jest.Mock;

  beforeEach(() =>
    mockOnFoodSelected = jest.fn()
  );

  describe('when the recipes are loading', () => {
    beforeEach(() =>
      wrapper = shallow(
        <FoodForBadge
          isLoading={false}
          currentBadgeFood={food}
          onFoodSelected={mockOnFoodSelected}>
          Children
        </FoodForBadge>
      )
    );

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());

    test('can click badge', () => {
      const onClick = wrapper.find(ImageGrid).first().props().onClick;
      if (onClick) {
        onClick('badgeId');
      }
      expect(mockOnFoodSelected).toHaveBeenCalledWith('badgeId');
    });
  });

  describe('when the food is not loaded', () => {
    beforeEach(() =>
      wrapper = shallow(
        <FoodForBadge
          isLoading={true}
          currentBadgeFood={food}
          onFoodSelected={mockOnFoodSelected}>
          Children
        </FoodForBadge>
      )
    );

    test('does not render anything', () =>
      expect(wrapper.children().exists()).toBe(false));
  });

});
