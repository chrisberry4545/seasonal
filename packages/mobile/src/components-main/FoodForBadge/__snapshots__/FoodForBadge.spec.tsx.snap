// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoodForBadge /> when the recipes are loading renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 100,
    }
  }
>
  <TextMedium
    style={
      Object {
        "marginBottom": 12,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    Found in
  </TextMedium>
  <ImageGrid
    data={
      Array [
        Object {},
        Object {},
      ]
    }
    data-e2e="food-for-badge-grid"
    onClick={[MockFunction]}
  />
</View>
`;
